src/CarbonAware.WebApi/src/Models/EmissionsForecastBaseDTO.cs

```diff
@@ -13,14 +13,14 @@ public record EmissionsForecastBaseDTO
     /// </summary>
```

Comment:
I have concerns that this current implementation does not match the specification, particularly this `requestedAt` summary:

```
    /// For current requests, this value is the timestamp the request for forecast data was made.
    /// For historical forecast requests, this value is the timestamp used to access the most 
    /// recently generated forecast as of that time. 
```
The point of the `requestedAt` field is to enable the user to not need to know about WattTime details like when exactly forecasts are generated.  Instead, the intent is to have them specify the time that they would have made a request for a current forecast and let the Carbon Aware API manage retrieving the most recent forecast as of that timestamp.

I didn't see anything that leads me to believe are doing that.  Are there details that I am missing here?

Reply:
@bderusha , definitely the text has to be updated since now the user can specify a requestedAt. Given the fact this property is a required one and if the user doesn't specify one as part of the request (having a nullable requestedAt) the framework takes care of the BadRequest.

Reply:
@juzuluag I think we're getting our wires crossed.  It may be easier to chat synchronously about this.  It is purposefully included in the response payload for `/emissions/forecasts/current` and I think it should stay there. My concern is that I believe this implementation for historical forecasts would yield the following user experience:

POST `/emissions/forecasts/batch` with payload `{ "requestedAt": "2022-06-01T16:07:32Z", ... }` returns `400` because WattTime only has generated forecasts for `2022-06-01T16:05:00Z`,  `2022-06-01T16:10:00Z`, etc.

The specification states that "the timestamp used to access the *most recently* generated forecast as of that time." which would mean that

POST `/emissions/forecasts/batch` with payload `{ "requestedAt": "2022-06-01T16:07:32Z", ... }` returns the forecast generated at `2022-06-01T16:05:00Z`.  Since that is what the `/current` endpoint would have returned to any user making a LIVE request at exactly `2022-06-01T16:07:32Z`

Does that make sense? Happy to sync up on this, to make sure we get aligned here.

Reply:
> @juzuluag I think we're getting our wires crossed. It may be easier to chat synchronously about this. It is purposefully included in the response payload for `/emissions/forecasts/current` and I think it should stay there. My concern is that I believe this implementation for historical forecasts would yield the following user experience:
> 
> POST `/emissions/forecasts/batch` with payload `{ "requestedAt": "2022-06-01T16:07:32Z", ... }` returns `400` because WattTime only has generated forecasts for `2022-06-01T16:05:00Z`, `2022-06-01T16:10:00Z`, etc.
> 
> The specification states that "the timestamp used to access the _most recently_ generated forecast as of that time." which would mean that
> 
> POST `/emissions/forecasts/batch` with payload `{ "requestedAt": "2022-06-01T16:07:32Z", ... }` returns the forecast generated at `2022-06-01T16:05:00Z`. Since that is what the `/current` endpoint would have returned to any user making a LIVE request at exactly `2022-06-01T16:07:32Z`
> 
> Does that make sense? Happy to sync up on this, to make sure we get aligned here.

Sure we can sync. As a heads up, the request to wattitme succeeds regardless whether is exact or not. It returns the 'ceiling' date time of the requestedAt.

Reply:
Ah ha! That was the piece I was missing.  Thank you for clarifying :) That is a strange choice from WattTime IMO.  Can I make one small suggestion then to tweak this into alignment with the spec.  Right now, 

`POST /emissions/forecasts/batch` 
with payload `{ "requestedAt": "2022-06-01T16:07:32Z", ... }`
returns a forecast `{ "generatedAt": "2022-06-01T16:10:00Z" }` 

What the user expects is for that request to return a forecast `{ "generatedAt": "2022-06-01T16:05:00Z" }` because that was the available forecast at `16:07:32`,

We have a bit of "clever" math that we use to snap dates to their nearest 5 minute increment in the [WattTimeMocker](https://github.com/microsoft/carbon-aware-sdk/blob/dev/src/CarbonAware.WebApi/test/integrationTests/mocks/WattTimeDataSourceMocker.cs#L63-L66)

I think that if we use a variation of that to request the FLOOR rather than letting wattTime give us the CEILING then everything should work as expected

```
var d = TimeSpan.FromMinutes(5);
`var fiveMinuteFloorDate = new DateTimeOffset((requestedAt.Ticks / d.Ticks) * d.Ticks, TimeSpan.Zero);
```

Reply:
Plz take a look at the https://github.com/microsoft/carbon-aware-sdk/pull/110/commits/a892f3f5e4c68c874082f771bed731c868c96959 , since I decoupled the classes to clarify what is the request data vs returned data.
For the remaining work (set the requestetAt date to FLOOR, instead of wattime to decide), we will handle in a different push.